crates/tests/agile_reference/tests/tests.rs

```diff
@@ -0,0 +1,15 @@
+use windows::core::AgileReference;
+use windows::Media::Control::GlobalSystemMediaTransportControlsSessionManager;
+
+#[test]
+fn test() {
+    let manager = GlobalSystemMediaTransportControlsSessionManager::RequestAsync().unwrap().get().unwrap();
+    let reference = AgileReference::<GlobalSystemMediaTransportControlsSessionManager>::new(&manager).unwrap();
```

Comment:
```suggestion
    let reference = AgileReference::new(&manager).unwrap();
```

Will this work?

Reply:
This won't work. `T` and `U` don't have any relationship to Rust right now.

Reply:
This seems to work:

```rust
/// A type representing an agile reference to a COM/WinRT object.
#[repr(transparent)]
#[derive(Clone, PartialEq, Eq)]
pub struct AgileReference<T>(IAgileReference, PhantomData<T>);

impl<T: Interface> AgileReference<T> {
    /// Creates an agile reference to the object.
    pub fn new(object: &T) -> Result<Self> {
        unsafe {
            let object: &IUnknown = std::mem::transmute(object);
            RoGetAgileReference(AGILEREFERENCE_DEFAULT, &T::IID, object).map(|reference| Self(reference, Default::default()))
         }
    }

    /// Retrieves a proxy to the target of the `AgileReference` object that may safely be used within any thread context in which get is called.
    pub fn resolve(&self) -> Result<T> {
        unsafe { self.0.Resolve() }
    }
}

unsafe impl<T: Interface> Send for AgileReference<T> {}
unsafe impl<T: Interface> Sync for AgileReference<T> {}

fn main() -> Result<()> {
    let manager = GlobalSystemMediaTransportControlsSessionManager::RequestAsync()?.get()?;
    let reference = AgileReference::new(&manager)?;

    let handle = std::thread::spawn(move || {
        let manager = reference.resolve().unwrap();
        manager.GetSessions().unwrap();
    });
    handle.join().unwrap();
    Ok(())
}
```

Reply:
This would be pretty much the same as [it was before](https://github.com/microsoft/windows-rs/pull/1474/commits/ee1aa495970340a7b10ff9ec58d8622a28b4f19c#diff-753f3bff5e476266fac3a79368ba3d184669e90cdc72acd7374fdcfde54aeac4L15-R13), only that it was using `&'a T: IntoParam<'a, IUnknown>` insted of transmuting. I'm not sure if transmuting here is a good idea since there's no constraint that `T` is `repr(transparent)` around `IUnknown`. In my understanding, using `IntoParam` here would result in the same behaviour.

Reply:
Sure, either way should be fine. Just as long as the inference is preserved. I didn't realize that would be lost with my earlier suggestion.